### PR TITLE
OF-2156: Fix for BOSH async issues

### PR DESCRIPTION
**Pull request by guusdk**

### CODE REVIEW

**Comment by guusdk** [comment]
This PR includes the change from https://github.com/igniterealtime/Openfire/pull/1761 as I'm hoping that the changes provided in the later commit fix the issue exposed by the former.

**Comment by Fishbowler** [comment]
There's some conflicts following the merge of #1761. There's no thread.sleep, and you've upped the connections to 50 consecutive connections. This is intentional? This PR contains the intended state of these tests?

**Comment by Fishbowler** [comment]
nm. seen the commit history :)

File: xmppserver/src/main/java/org/jivesoftware/openfire/http/HttpConnection.java [hunk 1]
```diff
@@ -113,10 +143,17 @@ public void deliverBody(String body, boolean async) throws HttpConnectionClosedE
             isClosed = true;
         }
 
-        if (body == null) {
-            body = getSession().createEmptyBody(false);
-        }
-        HttpBindServlet.respond(getSession(), this.context, body, async);
+        HttpBindServlet.respond(getSession(), this.context, body != null ? body : session.createEmptyBody(false), async);
```
Comment (Fishbowler) [inline xmppserver/src/main/java/org/jivesoftware/openfire/http/HttpConnection.java:146]: Why the mix of session and getSession() here? Are they equivalent here?

**Review by Fishbowler — APPROVED** [review]
Refactoring looks sound. I can't validate the deep side of the business logic, but I'm convinced that if the tests pass that the intended problem is solved.

**Pull request merged**